app/models/host_report.rb

```diff
@@ -56,6 +57,19 @@ def self.search_by_keyword(_key, operator, value)
     }
   end
 
+  def self.summarise(_time = 1.day.ago, *hosts, format)
+    list = {}
+    raise ::Foreman::Exception, N_("invalid host list") unless hosts
+    hosts.flatten.each do |host|
+      host_report_id = HostReport.where(host_id: host.id, format: HostReport.formats[format]).maximum(:id)
+      next if host_report_id.nil?
+      report = HostReport.find(host_report_id)
```

Comment:
This will give you what you need: `report = host.last_host_report_object` so you can replace all the three lines with this single statement. See app/models/concerns/foreman_host_reports/host_extensions.rb.